web/src/beta/features/Editor/Story/PagesPanel/index.tsx

```diff
@@ -1,135 +1,103 @@
-            }}
-            renderItem={(storyPage, i) => {
-              const title = (getFieldValue(storyPage.property.items ?? [], "title", "title") ??
-                t("Untitled")) as string;
-              const hasEmptySpace = isEmptyString(title);
-              return (
-                <PageItemWrapper
-                  key={storyPage.id}
-                  pageCount={i + 1}
-                  isSwipeable={storyPage.swipeable}>
-                  <PageItem
-                    key={i}
-                    isSelected={selectedStoryPage?.id === storyPage.id}
-                    isOpenAction={openedPageId === storyPage.id}
-                    onItemClick={() => handleStoryPageSelect(storyPage.id)}
-                    onActionClick={() => setOpenedPageId(old => (old ? undefined : storyPage.id))}
-                    onOpenChange={isOpen => {
-                      setOpenedPageId(isOpen ? storyPage.id : undefined);
-                    }}
-                    onPageDelete={() => handleStoryPageDelete(storyPage.id)}
-                    title={hasEmptySpace || !title ? t("Untitled") : title}
-                    setOpenedPageId={setOpenedPageId}
-                    propertyId={storyPage.property.id}
-                    storyPage={storyPage}
-                    onPropertyUpdate={handlePropertyValueUpdate}
-                  />
-                </PageItemWrapper>
-              );
-            }}
-          />
-        </SContentUp>
-        <SContentBottom>
-          <Action
-            icon="square"
-            title={`+ ${t("New Page")}`}
-            onClick={() => handleStoryPageAdd(false)}
-          />
-          {/* <Action icon="swiper" title={`+ ${t("New Swipe")}`} onClick={() => onPageAdd(true)} /> */}
-        </SContentBottom>
-      </SContent>
+      <ButtonWrapper>
+        <Button
+          icon="plus"
+          title="New Page"
+          size="small"
+          extendWidth
+          onClick={() => handleStoryPageAdd(false)}
+        />
+      </ButtonWrapper>
+      <Wrapper onScroll={openedPageId ? () => setOpenedPageId(undefined) : undefined}>
+        <DragAndDropList
+          items={DraggableStoryPageItems}
+          handleClassName={PAGES_DRAG_HANDLE_CLASS_NAME}
+          onMoveEnd={handleMoveEnd}
+          onMoveStart={handleMoveStart}
+          dragDisabled={false}
+        />
+        <EmptySpace onClick={() => handleStoryPageSelect(undefined)} />
+      </Wrapper>
     </Panel>
   );
 };
 
 export default PagesPanel;
 
-const SContent = styled.div`
-  display: flex;
-  flex-direction: column;
-  height: 100%;
-`;
-
-const SContentUp = styled.div`
-  flex-grow: 1;
-  height: 0;
-  overflow-y: auto;
-  ::-webkit-scrollbar {
-    display: none;
-  }
+const Wrapper = styled("div")(({ theme }) => ({
+  display: "flex",
+  flexDirection: "column",
+  height: "100%",
+  gap: theme.spacing.normal,
+  overflowY: "auto",
+  boxSizing: "border-box",
+  "::-webkit-scrollbar": {
+    display: "none",
+  },
```

Comment:
```suggestion
```
I'm not quite sure why we need this, but for current visualizer we want the scrollbar shows up when it's scrollable.